addons/sale/models/sale.py

```diff
@@ -1092,7 +1092,7 @@ def _get_real_price_currency(self, product, rule_id, qty, uom, pricelist_id):
             if currency_id.id == product_currency.id:
                 cur_factor = 1.0
             else:
-                cur_factor = currency_id._get_conversion_rate(product_currency, currency_id)
+                cur_factor = currency_id._get_conversion_rate(product_currency, currency_id, self.company_id, self.order_id.date_order)
```

Comment:
If my reading is correct, _get_conversion_rate takes an `id` (int) not a res.company (recordset).

Reply:
my suggestion is: change the API so that it takes a recordset